### PR TITLE
[draft] To add brief CPU optimization intro in compile/index.rst

### PR DESCRIPTION
**Pull request by ZailiWang**
@jingxu10 @ZhaoqiongZ @jgong5 @mingfeima and others, plz help review, thanks.

### CODE REVIEW

File: docs/source/compile/index.rst [hunk 1]
```diff
@@ -51,6 +51,13 @@ You can get the full list of community backends by running :func:`~torch._dynamo
 
     compile
 
+Compiled workloads on modern CPUs are usually optimized by SIMD instruction sets. SIMD (Single Instruction Multiple Data) is a typical parallel processing technique for high performance computing, such as deep learning model training and inference. With SIMD applied, each compute unit performs the same instruction with different allocated data at any given timeslot. The most commonly deployed x86 instruction set architectures (ISAs) enabling SIMD include `AVX, AVX2, AVX-512 <https://en.wikipedia.org/wiki/Advanced_Vector_Extensions>` and `AMX <https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions>`.
+
+Which ISAs are supported by a machine can be checked with `the tutorial <https://linuxconfig.org/identify-if-my-cpu-using-32-bit-or-64-bit-instruction-set>`.
+
+Specifically, with a server having AMX instructions enabled, workloads performance can be further boosted by `leveraging AMX <https://pytorch.org/tutorials/recipes/recipes/amx.html>`.
```
Comment (jgong5) [inline docs/source/compile/index.rst:58]: Can we find a better place to put this which is more a backend-specific intro? Here, the `Optimizations` section talks more about the frontend API to list backend options.

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: docs/source/compile/index.rst (hunk 1 above)
Comment (ZailiWang) [reply docs/source/compile/index.rst:58]: Hi @Guobing-Chen please check Jiong's suggestion, what would you like to comment?

Comment (Guobing-Chen) [reply docs/source/compile/index.rst:58]: Make sense that this is more backend (especially CPU) specific. The problem is there is no backend specific page now.

Comment (ZailiWang) [reply docs/source/compile/index.rst:58]: Can we seperate the `optimizations` section into sub-sections like `frontend APIs`, `CPU backend`, and the added part is under `CPU backend`? And later other peers can contribute intro of their respective backends, like `CUDA backend` or `MPS backend` etc.

**Comment by malfet** [comment]
Please keep draft PRs draft.

File: docs/source/compile/index.rst (hunk 1 above)
Comment (jgong5) [reply docs/source/compile/index.rst:58]: I think the "backend" here for `torch.compile` is the dynamo backend, not the "inductor" backend like "CPU backend" we are talking about here. I guess a place talking about inductor backends would be more appropriate?

**Review by jgong5 — COMMENTED** [review]
@mingfeima

**Comment by ZailiWang** [comment]
As discussed, it's not appropriate to add CPU backend optim mechanism at this page. Close this PR and will create a new one.

**Pull request closed without merging**